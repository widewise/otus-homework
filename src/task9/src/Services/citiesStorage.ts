import {ICityWeather} from "../Models/weather";
import {Error} from "../components/Error";

const CITIES_KEY = "CITIES";

function getCities(): ICityWeather[] {
    const jsonData = localStorage.getItem(CITIES_KEY);
    if (!jsonData) {
        throw Error("Storage is not initialized");
    }
    return  JSON.parse(jsonData) as ICityWeather[];
}

export const citiesStorage = {
    init: (cities: ICityWeather[]) => {
        localStorage.setItem(CITIES_KEY, JSON.stringify(cities));
    },

    searchCities: (search: string = ""): ICityWeather[] => {
        return getCities().filter(c => c.name.includes(search));
    },

    getCity: (id: string): ICityWeather | undefined => {
        const city = getCities().find(c => c.id === id);
        if(!city) {
            console.log(`City with id ${id} doesn't exist`);
            return;
        }
        return city;
    }
}