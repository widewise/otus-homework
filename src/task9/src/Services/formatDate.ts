const padTo2Digits = (num: number) => {
    return num.toString().padStart(2, '0');
}

export function formatDate(date: Date): string {
    const aDate = new Date(date);
    if(!aDate) {
        return "";
    }

    return `${padTo2Digits(aDate.getDate())}:${padTo2Digits(aDate.getMonth() + 1)}:${aDate.getFullYear()}`;
}