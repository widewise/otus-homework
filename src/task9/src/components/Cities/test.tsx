import React from 'react';
import renderer from "react-test-renderer";
import { Cities } from "./index";
import { useSearchParams } from "react-router-dom";
import { citiesStorage } from "../../Services/citiesStorage";

jest.mock("react-router-dom");
jest.mock("../../Services/citiesStorage");

describe(`${Cities.name}`, () =>{
    let actual: any;
    describe(`rerender ${Cities.name}`, () => {
        beforeEach(() => {
            //arrange
            (useSearchParams as jest.Mock).mockImplementationOnce(() => [{
                get: (): string => "1"
            }]);
            (citiesStorage.searchCities as jest.Mock).mockImplementationOnce(() => []);

            //act
            actual = renderer.create(<Cities />).toJSON();
        });

        it("should match snapshot", () =>{
            //assert
            expect(actual).toMatchSnapshot();
        });
    });
});