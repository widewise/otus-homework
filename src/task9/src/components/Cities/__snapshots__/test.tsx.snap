// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities rerender Cities should match snapshot 1`] = `
<div>
  <div
    className="cities-search"
  >
    <label>
      City:
    </label>
    <input
      onChange={[Function]}
      placeholder="Enter city"
      type="text"
      value="1"
    />
  </div>
  <ul />
</div>
`;
